static/js/components/dashboard/CourseSubRow.js

```diff
@@ -109,25 +118,42 @@ export default class CourseSubRow extends React.Component {
       addCourseEnrollment
     } = this.props;
 
-    if (courseRun.status === STATUS_OFFERED) {
-      let enrollStartDate = courseRun.enrollment_start_date ? moment(courseRun.enrollment_start_date) : null;
-      if (isCurrentlyEnrollable(enrollStartDate, now)) {
-        return <CourseAction
-          courseRun={courseRun}
-          coursePrice={coursePrice}
-          checkout={checkout}
-          now={now}
-          hasFinancialAid={hasFinancialAid}
-          financialAid={financialAid}
-          openFinancialAidCalculator={openFinancialAidCalculator}
-          addCourseEnrollment={addCourseEnrollment}
-        />;
-      }
+    let enrollStartDate = courseRun.enrollment_start_date ? moment(courseRun.enrollment_start_date) : null;
+    if (isCurrentlyEnrollable(enrollStartDate, now)) {
+      return <CourseAction
+        courseRun={courseRun}
+        coursePrice={coursePrice}
+        checkout={checkout}
+        now={now}
+        hasFinancialAid={hasFinancialAid}
+        financialAid={financialAid}
+        openFinancialAidCalculator={openFinancialAidCalculator}
+        addCourseEnrollment={addCourseEnrollment}
+      />;
     }
-    return <div className="course-action">
-      { courseRun.status === STATUS_NOT_PASSED ? 'Failed' : '' }
-    </div>;
-  }
+  };
+
+  courseRunStatus = (courseRun: CourseRun) => {
+    if (courseRun.status === STATUS_NOT_PASSED ) {
+      return 'Failed';
+    } else if ( courseRun.status === STATUS_PASSED ) {
+      return 'Passed';
+    }
+  };
+
+  renderCourseRunGrade = (courseRun: CourseRun) => (
```

Comment:
this isn't really returning a course run _grade_. just a string indicating the status. i think this should be called `renderCourseRunStatus`. the contents of `courseRunStatus` (line 136) can probably just be cut/pasted into the body of this one